--- a/Makefile	2024-02-16 14:58:29.220126035 +0000
+++ b/Makefile	2024-02-16 14:58:46.572468923 +0000
@@ -393,7 +393,7 @@
 
 KBUILD_AFLAGS   := -D__ASSEMBLY__
 KBUILD_CFLAGS   := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
-		   -fno-strict-aliasing -fno-common -fshort-wchar \
+		   -fno-strict-aliasing -fcommon -fshort-wchar \
 		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
 		   -std=gnu89
