diff --git a/arch/arm/boot/dts/sun8i-v3s-funkey.dts b/arch/arm/boot/dts/sun8i-v3s-funkey.dts
index 0a8ce5c0..31fddd2e 100644
--- a/arch/arm/boot/dts/sun8i-v3s-funkey.dts
+++ b/arch/arm/boot/dts/sun8i-v3s-funkey.dts
@@ -81,13 +81,14 @@
 &i2c0 {
 	status = "okay";
 
-	/* axp209: pmic@34 {
+	axp209: pmic@34 {
 		compatible = "x-powers,axp209";
 		reg = <0x34>;
-		interrupts = <0>;
+		interrupt-parent = <&pio>;
+		interrupts = <1 5 IRQ_TYPE_EDGE_FALLING>;
 		interrupt-controller;
 		#interrupt-cells = <1>;
-	}; */
+	};
 };
 
 &mmc0 {
@@ -104,9 +105,24 @@
 	status = "okay";
 };
 
-/*
 #include "axp209.dtsi"
 
+&ac_power_supply {
+	status = "okay";
+};
+
+&axp_gpio {
+	status = "okay";
+};
+
+&battery_power_supply {
+	status = "okay";
+};
+
+&usb_power_supply {
+	status = "okay";
+};
+
 &reg_dcdc2 {
 	regulator-always-on;
 	regulator-min-microvolt = <1000000>;
@@ -133,7 +149,6 @@
 	regulator-max-microvolt = <3300000>;
 	regulator-name = "avcc-pll";
 };
-*/
 
 &reg_vcc3v0 {
 	regulator-always-on;
diff --git a/drivers/regulator/axp20x-regulator.c b/drivers/regulator/axp20x-regulator.c
index 376a99b7..0718d191 100644
--- a/drivers/regulator/axp20x-regulator.c
+++ b/drivers/regulator/axp20x-regulator.c
@@ -169,7 +169,7 @@ static const struct regulator_desc axp20x_regulators[] = {
 		 AXP20X_DCDC2_V_OUT, 0x3f, AXP20X_PWR_OUT_CTRL, 0x10),
 	AXP_DESC(AXP20X, DCDC3, "dcdc3", "vin3", 700, 3500, 25,
 		 AXP20X_DCDC3_V_OUT, 0x7f, AXP20X_PWR_OUT_CTRL, 0x02),
-	AXP_DESC_FIXED(AXP20X, LDO1, "ldo1", "acin", 1300),
+	AXP_DESC_FIXED(AXP20X, LDO1, "ldo1", "acin", 3300),
 	AXP_DESC(AXP20X, LDO2, "ldo2", "ldo24in", 1800, 3300, 100,
 		 AXP20X_LDO24_V_OUT, 0xf0, AXP20X_PWR_OUT_CTRL, 0x04),
 	AXP_DESC(AXP20X, LDO3, "ldo3", "ldo3in", 700, 3500, 25,
