From ff9cbaaeb217a6cf2e2fffde614abfdfb3f6cf5d Mon Sep 17 00:00:00 2001
From: vincent <vbusoenseirb@gmail.com>
Date: Tue, 19 Nov 2019 14:02:56 +0100
Subject: [PATCH] changed st7789v gamma

---
 drivers/staging/fbtft/fb_st7789v.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fbtft/fb_st7789v.c b/drivers/staging/fbtft/fb_st7789v.c
index 00756de..212836f 100755
--- a/drivers/staging/fbtft/fb_st7789v.c
+++ b/drivers/staging/fbtft/fb_st7789v.c
@@ -26,8 +26,8 @@
 #define DRVNAME "fb_st7789v"
 
 #define DEFAULT_GAMMA \
-	"70 2C 2E 15 10 09 48 33 53 0B 19 18 20 25\n" \
-	"70 2C 2E 15 10 09 48 33 53 0B 19 18 20 25"
+	"f0 19 1e 0A 09 15 3D 44 51 19 14 13 2c 31\n" \
+	"f0 18 1E 0A 09 25 3F 43 52 19 14 13 2c 31"
 
 /**
  * enum st7789v_command - ST7789V display controller commands
-- 
1.9.1

