From 4e80937d23c58478e8a3ffe3a5cdb4114497c638 Mon Sep 17 00:00:00 2001
From: vincent <vbusoenseirb@gmail.com>
Date: Tue, 29 Sep 2020 08:32:44 +0200
Subject: [PATCH] increasd buffer size for fb_tft_notif zone to 400

---
 drivers/staging/fbtft/fbtft.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fbtft/fbtft.h b/drivers/staging/fbtft/fbtft.h
index 8a9daee..4638ad9 100644
--- a/drivers/staging/fbtft/fbtft.h
+++ b/drivers/staging/fbtft/fbtft.h
@@ -28,7 +28,7 @@
 #define FBTFT_GAMMA_MAX_VALUES_TOTAL 	128
 #define FBTFT_OVERLAY_NB_VALUES			4
 
-#define FBTFT_NOTIF_MAX_SIZE			256
+#define FBTFT_NOTIF_MAX_SIZE			400
 
 #define FBTFT_OF_INIT_CMD				BIT(24)
 #define FBTFT_OF_INIT_DELAY				BIT(25)
-- 
1.9.1

