 sun8i-v3s-licheepi-zero-dock.dts |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/sun8i-v3s-licheepi-zero-dock.dts b/arch/arm/boot/dts/sun8i-v3s-licheepi-zero-dock.dts
index 58c4e72..6e9dfcc 100644
--- a/arch/arm/boot/dts/sun8i-v3s-licheepi-zero-dock.dts
+++ b/arch/arm/boot/dts/sun8i-v3s-licheepi-zero-dock.dts
@@ -49,9 +49,9 @@
 	compatible = "licheepi,licheepi-zero-dock", "licheepi,licheepi-zero",
 		     "allwinner,sun8i-v3s";
 
-	aliases {
+	/*aliases {
 		ethernet0 = &emac;
-	};
+	};*/
 
 	leds {
 		/* The LEDs use PG0~2 pins, which conflict with MMC1 */
@@ -68,9 +68,9 @@
 	status = "okay";
 };
 
-&emac {
+/*&emac {
 	status = "okay";
-};
+};*/
 
 &mmc1 {
 	broken-cd;
