 sun8i-v3s-licheepi-zero.dts |   18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-v3s-licheepi-zero.dts b/arch/arm/boot/dts/sun8i-v3s-licheepi-zero.dts
index d63a9a3..063c6bf 100644
--- a/arch/arm/boot/dts/sun8i-v3s-licheepi-zero.dts
+++ b/arch/arm/boot/dts/sun8i-v3s-licheepi-zero.dts
@@ -118,3 +118,21 @@
 	usb0_id_det-gpio = <&pio 5 6 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 };
+
+&spi0 {
+       status = "okay";
+
+       st7789v@0 {
+               compatible = "sitronix,st7789v";
+               reg = <0>;
+
+               spi-max-frequency = <24000000>;
+               txbuflen = <115200>;
+               rotate = <0>;
+               fps = <25>;
+               buswidth = <8>;
+               reset-gpios = <&pio 1 4 GPIO_ACTIVE_LOW>;
+               dc-gpios = <&pio 1 5 GPIO_ACTIVE_LOW>;
+               debug = <0>;
+       };
+};
