 sun8i-v3s.dtsi |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/sun8i-v3s.dtsi b/arch/arm/boot/dts/sun8i-v3s.dtsi
index 5379f2d..e437d3f 100644
--- a/arch/arm/boot/dts/sun8i-v3s.dtsi
+++ b/arch/arm/boot/dts/sun8i-v3s.dtsi
@@ -54,14 +54,14 @@
 		#size-cells = <1>;
 		ranges;
 
-		simplefb_lcd: framebuffer@0 {
+		/*simplefb_lcd: framebuffer@0 {
 			compatible = "allwinner,simple-framebuffer",
 				     "simple-framebuffer";
 			allwinner,pipeline = "de0-lcd0";
 			clocks = <&ccu CLK_BUS_TCON0>, <&display_clocks 0>,
 				 <&display_clocks 6>, <&ccu CLK_TCON0>;
 			status = "disabled";
-		};
+		};*/
 	};
 
 	cpus {
