Author: Andrea Veri <and@debian.org>
Description: Disable configure's run from the autogen file.

Index: agg-2.5+dfsg1/autogen.sh
===================================================================
--- a/autogen.sh	2007-10-11 00:06:16.000000000 +0200
+++ b/autogen.sh	2012-05-01 16:57:37.916862783 +0200
@@ -18,6 +18,6 @@
 automake --foreign --add-missing --ignore-deps
 
 # and finally invoke our new configure
-./configure $*
+[ -n "$NOCONFIGURE" ] || ./configure $*
 
 # end
