diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 469dd71..e91a8d3 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -281,6 +281,7 @@ dtb-$(CONFIG_MACH_SUN8I_H3) += \
 	sun8i-h3-orangepi-plus2e.dtb \
 	sun8i-h3-nanopi-neo.dtb
 dtb-$(CONFIG_MACH_SUN8I_V3S) += \
+	sun8i-v3s-funkey.dtb \
 	sun8i-v3s-licheepi-zero.dtb
 dtb-$(CONFIG_MACH_SUN50I) += \
 	sun50i-a64-pine64-plus.dtb \
