--- a/configure.in
+++ b/configure.in
@@ -39,7 +39,7 @@
 # used as platform library in examples:
 # todo, make the PREFERED_PLATFORM selectable, after the set of possible 
 # Platforms to link the examples have been evaluated.
-PREFERED_PLATFORM=X11
+PREFERED_PLATFORM=sdl
 case "$host" in
   *darwin* )
     OSX_LIBS="-framework Carbon -framework QuickTime"
@@ -120,9 +120,7 @@
 if test "$no_x" = "yes"; then
   AC_MSG_WARN([*** X11 not found! Omitting X11 layer.])
 fi
-AM_CONDITIONAL(ENABLE_X11,[test x$no_x = x -a xno != x$enable_platform -a x$win32_host != xyes])
-AC_SUBST(x_includes)
-AC_SUBST(x_libraries)
+AM_CONDITIONAL(ENABLE_X11,0)
 dnl ###############################################
 
 dnl Settung up library version

--- a/include/agg_renderer_outline_aa.h
+++ b/include/agg_renderer_outline_aa.h
@@ -1375,7 +1375,7 @@
         //---------------------------------------------------------------------
         void profile(const line_profile_aa& prof) { m_profile = &prof; }
         const line_profile_aa& profile() const { return *m_profile; }
-        line_profile_aa& profile() { return *m_profile; }
+//        line_profile_aa& profile() { return *m_profile; }
 
         //---------------------------------------------------------------------
         int subpixel_width() const { return m_profile->subpixel_width(); }
