--- a/board/sunxi/board.c	2017-08-12 11:24:24.000000000 +0200
+++ b/board/sunxi/board.c	2019-04-27 23:04:05.811352883 +0200
@@ -540,9 +540,9 @@
 	power_failed |= axp_set_sw(IS_ENABLED(CONFIG_AXP_SW_ON));
 #endif
 #endif
-	printf("DRAM:");
+	//printf("DRAM:");
 	ramsize = sunxi_dram_init();
-	printf(" %d MiB\n", (int)(ramsize >> 20));
+	//printf(" %d MiB\n", (int)(ramsize >> 20));
 	if (!ramsize)
 		hang();
 
--- a/common/console.c	2017-08-12 11:24:24.000000000 +0200
+++ b/common/console.c	2019-04-27 22:30:13.944154949 +0200
@@ -695,7 +695,7 @@
 	if (getenv("silent") != NULL)
 		gd->flags |= GD_FLG_SILENT;
 	else
-		gd->flags &= ~GD_FLG_SILENT;
+		gd->flags |= GD_FLG_SILENT;
 #endif
 }
 
--- a/common/spl/spl.c	2017-08-12 11:24:24.000000000 +0200
+++ b/common/spl/spl.c	2019-04-27 23:06:27.269590207 +0200
@@ -316,7 +316,7 @@
 		loader = spl_ll_find_loader(spl_boot_list[i]);
 #if defined(CONFIG_SPL_SERIAL_SUPPORT) && defined(CONFIG_SPL_LIBCOMMON_SUPPORT)
 		if (loader)
-			printf("Trying to boot from %s", loader->name);
+		  ;//printf("Trying to boot from %s", loader->name);
 		else
 			puts("SPL: Unsupported Boot Device!\n");
 #endif
@@ -407,8 +407,9 @@
 
 	gd->have_console = 1;
 
-	puts("\nU-Boot SPL " PLAIN_VERSION " (" U_BOOT_DATE " - " \
+	/*puts("\nU-Boot SPL " PLAIN_VERSION " (" U_BOOT_DATE " - "	\
 			U_BOOT_TIME ")\n");
+	*/
 #ifdef CONFIG_SPL_DISPLAY_PRINT
 	spl_display_print();
 #endif
